packages/astro/src/runtime/server/astro-island.ts

```diff
@@ -93,14 +93,15 @@ declare const Astro: {
 					}
 					this.start();
 				}
-				start() {
+				async start() {
 					const opts = JSON.parse(this.getAttribute('opts')!) as Record<string, any>;
 					const directive = this.getAttribute('client') as directiveAstroKeys;
 					if (Astro[directive] === undefined) {
 						window.addEventListener(`astro:${directive}`, () => this.start(), { once: true });
 						return;
 					}
-					Astro[directive]!(
+					try {
+					await Astro[directive]!(
```

Comment:
Is this change safe?

Reply:
What could be an issue?

Reply:
I'm genuinely asking 😂 sometimes we don't wait for promises on purpose

Reply:
The one thing I can think of is a "thenable". But if a custom client directive is using it in a fancy way, it is just asking for trouble.

```js
await { then(resolve) { resolve(5) } }
```

Reply:
Thank you!